gateway-shell/src/test/java/org/apache/knox/gateway/shell/table/KnoxShellTableFilterTest.java

```diff
@@ -50,18 +51,86 @@ public void testRegexFilter() throws KnoxShellTableFilterException {
+    }
+  }
+
+  @Test
+  public void testGreaterThanOrEqualToThanFilter() throws KnoxShellTableFilterException, ParseException {
+    assertEquals(TABLE.filter().name("Column Integer").greaterThanOrEqualTo(30).getRows().size(), 2);
+    assertEquals(TABLE.filter().name("Column Integer").greaterThanOrEqualTo(50).getRows().size(), 1);
+
+    assertEquals(TABLE.filter().name("Column Double").greaterThanOrEqualTo(40d).getRows().size(), 2);
+    assertEquals(TABLE.filter().name("Column Double").greaterThanOrEqualTo(20d).getRows().size(), 3);
+
+    assertEquals(TABLE.filter().name("Column String").greaterThanOrEqualTo("text50").getRows().size(), 2);
+    assertEquals(TABLE.filter().name("Column String").greaterThanOrEqualTo("text30").getRows().size(), 3);
+
+    synchronized (TEST_DATE_FORMAT) {
+      assertEquals(TABLE.filter().name("Column Date").greaterThanOrEqualTo(TEST_DATE_FORMAT.parse("2009-12-31")).getRows().size(), 3);
+      assertEquals(TABLE.filter().name("Column Date").greaterThanOrEqualTo(TEST_DATE_FORMAT.parse("2012-01-01")).getRows().size(), 2);
+    }
+  }
+
+  @Test
+  public void testLessThanOrEqualToThanFilter() throws KnoxShellTableFilterException, ParseException {
+    assertEquals(TABLE.filter().name("Column Integer").lessThanOrEqualTo(30).getRows().size(), 2);
+    assertEquals(TABLE.filter().name("Column Integer").lessThanOrEqualTo(50).getRows().size(), 3);
+
+    assertEquals(TABLE.filter().name("Column Double").lessThanOrEqualTo(40d).getRows().size(), 2);
+    assertEquals(TABLE.filter().name("Column Double").lessThanOrEqualTo(20d).getRows().size(), 1);
+
+    assertEquals(TABLE.filter().name("Column String").lessThanOrEqualTo("text50").getRows().size(), 2);
+    assertEquals(TABLE.filter().name("Column String").lessThanOrEqualTo("text30").getRows().size(), 1);
+
+    synchronized (TEST_DATE_FORMAT) {
+      assertEquals(TABLE.filter().name("Column Date").lessThanOrEqualTo(TEST_DATE_FORMAT.parse("2012-02-01")).getRows().size(), 3);
+      assertEquals(TABLE.filter().name("Column Date").lessThanOrEqualTo(TEST_DATE_FORMAT.parse("2010-01-01")).getRows().size(), 1);
+    }
+  }
+
+  @Test
+  public void testEqualToFilter() throws KnoxShellTableFilterException, ParseException {
+    assertEquals(TABLE.filter().name("Column Integer").equalTo(60).getRows().size(), 0);
+    assertEquals(TABLE.filter().name("Column Integer").equalTo(50).getRows().size(), 1);
+
+    assertEquals(TABLE.filter().name("Column Double").equalTo(40d).getRows().size(), 1);
+    assertEquals(TABLE.filter().name("Column Double").equalTo(90d).getRows().size(), 0);
+
+    assertEquals(TABLE.filter().name("Column String").equalTo("text90").getRows().size(), 0);
+    assertEquals(TABLE.filter().name("Column String").equalTo("text30").getRows().size(), 1);
+
+    synchronized (TEST_DATE_FORMAT) {
+      assertEquals(TABLE.filter().name("Column Date").equalTo(TEST_DATE_FORMAT.parse("2012-02-01")).getRows().size(), 1);
```

Comment:
Aren't there 2 rows with `2012-01-01`?

Reply:
The date on line 42 is `2012-02-01` which is what I'm filtering for on line 132.

Reply:
Ahh...sorry...my mistake